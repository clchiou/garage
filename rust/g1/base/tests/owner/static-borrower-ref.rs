struct Struct<'a>(&'a [u8]);

impl<'a> TryFrom<&'a [u8]> for Struct<'a> {
    type Error = ();

    fn try_from(_: &'a [u8]) -> Result<Self, Self::Error> {
        std::unreachable!()
    }
}

g1_base::define_owner!(Owner for Struct);

fn static_borrow_ref(_: &Struct<'static>) {}

fn main() {
    let x = Owner::try_from(Vec::new()).unwrap();
    static_borrow_ref(x.deref());
}
