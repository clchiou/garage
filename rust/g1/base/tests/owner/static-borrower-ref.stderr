error[E0597]: `x` does not live long enough
  --> tests/owner/static-borrower-ref.rs:17:23
   |
16 |     let x = Owner::try_from(Vec::new()).unwrap();
   |         - binding `x` declared here
17 |     static_borrow_ref(x.deref());
   |                       ^--------
   |                       |
   |                       borrowed value does not live long enough
   |                       argument requires that `x` is borrowed for `'static`
18 | }
   | - `x` dropped here while still borrowed
